xbmc/pvr/epg/EpgSearchFilter.cpp

```diff
@@ -114,7 +117,7 @@ bool CPVREpgSearchFilter::MatchDuration(const CPVREpgInfoTagPtr &tag) const
 
 bool CPVREpgSearchFilter::MatchStartAndEndTimes(const CPVREpgInfoTagPtr &tag) const
 {
-  return (tag->StartAsLocalTime() >= m_startDateTime && tag->EndAsLocalTime() <= m_endDateTime);
+  return (tag->StartAsLocalTime() >= GetStartDateTime() && tag->EndAsLocalTime() <= GetEndDateTime());
```

Comment:
Maybe it's best to dump all the "cuteness" from the search filter class and to set the right start + stop values from outside (e.g. on search dialog init), before doing a search. What do you think?